File: TestModels/Extern/runtimes/python/src/simple_dafnyextern/extern/ExternConstructor.py
from simple_dafnyextern.internaldafny.ExternConstructor import *
import simple_dafnyextern_internaldafny_types
import Wrappers
import _dafny

class ExternConstructorClass:

  def __init__(self, input):
    if (_dafny.string_of(input) == "Error"):
      raise Exception("Python Constructor Exception")
    self.inputVal = input

  def GetValue(self):
    return Wrappers.Result_Success(self.inputVal)

  @staticmethod
  def Build(input):
      try:
        return Wrappers.Result_Success(ExternConstructorClass(input))
      except Exception as e:
        return Wrappers.Result_Failure(simple_dafnyextern_internaldafny_types.Error_Opaque(e))

File: TestModels/Extern/runtimes/go/ManualChanges/src/ExternConstructor/ExternConstructor.go
package ExternConstructor

import (
	"Wrappers"
	"dafny"
	"simpledafnyexterninternaldafnytypes"
	"types"
)

type ExternConstructorClass struct {
	Value *string
}

func (e ExternConstructorClass) GetValue() Wrappers.Result {
	return Wrappers.Companion_Result_.Create_Success_(dafny.SeqOfChars([]dafny.Char(*e.Value)...))
}

type CompanionStruct_ExternConstructorClass_ struct {
}

var Companion_ExternConstructorClass_ = CompanionStruct_ExternConstructorClass_{}

func (CompanionStruct_ExternConstructorClass_) Build(input dafny.Sequence) Wrappers.Result {
	if input.Equals(dafny.SeqOfChars([]dafny.Char("Error")...)) {
		return Wrappers.Companion_Result_.Create_Failure_(
			simpledafnyexterninternaldafnytypes.Companion_Error_.Create_Opaque_(types.OpaqueError{
				"Exception"}))
	}
	return Wrappers.Companion_Result_.Create_Success_(&ExternConstructorClass{func() *string {
		var s string
		for i := dafny.Iterate(input); ; {
			val, ok := i()
			if !ok {
				return &[]string{s}[0]
			} else {
				s = s + string(val.(dafny.Char))
			}
		}
	}()})
}
